Sources/FoundationEssentials/String/String+Internals.swift

```diff
@@ -138,20 +138,20 @@ extension String {
             return try block(buffer.baseAddress!)
         }
         #else
-        var iter = self.utf8.makeIterator()
 #if os(Windows)
+        var iter = self.utf8.makeIterator()
         let bLeadingSlash = if iter.next() == ._slash, iter.next()?.isLetter ?? false, iter.next() == ._colon { true } else { false }
-#else
-        let bLeadingSlash = false
-#endif
-
         // Strip the leading `/` on a RFC8089 path (`/[drive-letter]:/...` ).  A
         // leading slash indicates a rooted path on the drive for the current
         // working directory.
-        return try Substring(self.utf8.dropFirst(bLeadingSlash ? 1 : 0)).withCString {
+        return try Substring(self.utf8.dropFirst(bLeadingSlash ? 1 : 0)).replacing("/", with: "\\").withCString {
```

Comment:
Would it be better to adjust `appendingPathComponent` to use the `\` separator instead of the `/` separator, or do you think it's better that we always work with `/` separators until converting to a file system representation like we do here? (cc @jrflat since the former might update `URL`'s behavior I think)

Reply:
I think we do want to use `/` in `URL` up until converting to FSR, where we then convert to `\` on Windows. The `URL` APIs such as `.appending(path:)` handle inputs containing either `/` or `\` for Windows and standardizes them to `/`.

Reply:
Ok cool thanks for the context - yeah then this change makes sense to me to only swap the separators when going to a file system representation. Looks like CI hit some build failures but the approach to the change seems fine to me then